### DISCUSSION TITLE
A question about equivTo

### DISCUSSION THREAD

**Discussion by cshezhang** [soot-oss/SootUp#831] [Q&A] [answered]
Hi, I have a small question about `equivTo`. I am curious why not directly override `equals` and `hashCode`. Thank you!

**Comment by swissiety** [comment, marked as answer]
Hi,
because of the domain logic - e.g. with `Local`s its that the name inside a method is unique, but the type is not considered in the comparison - but if you want to compare the type as well you could extend it/compare via equivTo.
